deploy/cdk/lib/manifest-pipeline/manifest-pipeline-stack.ts

```diff
@@ -0,0 +1,713 @@
+            HttpMethods.HEAD,
+          ],
+          allowedOrigins: [
+            "*",
+          ],
+        },
+      ],
+      serverAccessLogsBucket: props.foundationStack.logBucket,
+      serverAccessLogsPrefix: 's3/data-broker/',
+    })
+
+    /* Manifest Bucket Policy */
+    this.manifestBucket.addToResourcePolicy(new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["s3:GetBucketCors"],
+      resources: [this.manifestBucket.bucketArn],
+      principals: [new CanonicalUserPrincipal(originAccessId.cloudFrontOriginAccessIdentityS3CanonicalUserId)],
+    }),
+    )
+
+    this.manifestBucket.addToResourcePolicy(new PolicyStatement({
+      effect: Effect.ALLOW,
+      actions: ["s3:GetObject"],
+      resources: [this.manifestBucket.bucketArn + '/*'],
+      principals: [new CanonicalUserPrincipal(originAccessId.cloudFrontOriginAccessIdentityS3CanonicalUserId)],
+    }),
+    )
+
+
+    // Add parameter store values for later reference by other apps
+    new StringParameter(this, 'sSMImageServerBaseUrl', {
+      type: ParameterType.STRING,
+      parameterName: `${appConfigPath}/image-server-base-url`,
+      stringValue: StringParameter.fromStringParameterAttributes(this, 'SSMImageServerHostname', { parameterName: props.imageServerHostname }).stringValue,
+      description: 'Image server base url',
+    })
+
+    new StringParameter(this, 'sSMImageSourceBucket', {
+      type: ParameterType.STRING,
+      parameterName: `${appConfigPath}/image-server-bucket`,
+      stringValue: props.foundationStack.publicBucket.bucketName,
+      description: 'Image source bucket',
+    })
+
+    new StringParameter(this, 'sSMManifestServerBaseUrl', {
+      type: ParameterType.STRING,
+      parameterName: `${appConfigPath}/manifest-server-base-url`,
+      stringValue: props.hostnamePrefix + '.' + props.domainName,
+      description: 'Manifest Server URL',
+    })
+
+    new StringParameter(this, 'sSMManifestBucket', {
+      type: ParameterType.STRING,
+      parameterName: `${appConfigPath}/manifest-server-bucket`,
+      stringValue: this.manifestBucket.bucketName,
+      description: 'S3 Bucket to hold Manifests',
+    })
+
+    new StringParameter(this, 'sSMProcessBucket', {
+      type: ParameterType.STRING,
+      parameterName: `${appConfigPath}/process-bucket`,
+      stringValue: this.processBucket.bucketName,
+      description: 'S3 Bucket to accumulate assets during processing',
+    })
+
+    new StringParameter(this, 'sSMRBSCS3ImageBucketName', {
+      type: ParameterType.STRING,
+      parameterName: `${appConfigPath}/rbsc-image-bucket`,
+      stringValue: props.rBSCS3ImageBucketName,
+      description: 'Name of the RBSC Image Bucket',
+    })
+
+    if (props.useExistingDnsZone) {
+      ManifestPipelineStack.hostedZoneToUse = HostedZone.fromLookup(this, 'HostedZone', { domainName: props.domainName })
+    } else {
+      ManifestPipelineStack.hostedZoneToUse = props.foundationStack.hostedZone
+    }
```

Comment:
I'm not sure this branching is needed anymore. The foundationStack already takes this prop, so by referencing `props.foundationStack.hostedZone`, this should point to the correct thing regardless of whether you are making a new foundation or importing hosted zone from an existing stack.

Reply:
I'll have to double-check with Justin.  I actually added this because of a comment he added to the foundation class:

 /**
   * If given, it will use the given Route53 Zone for this Vpc/DomainName
   * instead of creating one.
   *
   * Note: When using this option, this stack will not share the zone properties
   * via export/import to dependent stacks. Dependent stacks will need to
   * also be deployed with the same option used when the FoundationStack was
   * deployed.
   */
  readonly useExistingDnsZone?: boolean;

Reply:
That's what context does for us. ;) Thanks to the `env` context value, that will specify the hostedzone to use for all dependent stacks. The values from when the foundation stack is deployed get cached in `cdk.context.json` so in essence `foundationStack.hostedZone` references those values.

Reply:
I made these recommended changes.  Thanks.